### PR TITLE
feature request: allow searching for users by phone number

### LINKED ISSUE #895: Allow searching for users by phone number
**Is your feature request related to a problem? Please describe:**
Currently we are limited to searching the list of users by username. I want to be able to filter users by phone number similarly to how search by label on services is implemented.

**Describe the solution you'd like:**
Add support to search by phone number via filter drop down and search text field.

### PR DESCRIPTION
**Pull request by Quynhd07**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**

This PR allows for user search by phone number via filter drop down and search text field. 
Close #895

### CODE REVIEW

File: web/src/app/lists/QueryList.tsx [hunk 1]
```diff
@@ -76,6 +76,7 @@ export interface QueryListProps extends ControlledPaginatedListProps {
   // note: The `input.search` and `input.first` parameters are included by default, but can be overridden
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   variables?: any
+  mapVariables?: (vars: any) => any
```
Comment (mastercactapus) [inline web/src/app/lists/QueryList.tsx:79]: Need to add the correct type (`OperationVariables`) and add a comment for what this prop does.

File: web/src/app/users/UserPhoneNumberFilterContainer.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          label='Search by Phone Number'
+          helperText='Please provide your country code e.g. +1 (USA)'
+          type='tel'
+        />
+      </Grid>
+    </FilterContainer>
+  )
+}
+
+UserPhoneNumberFilterContainer.propTypes = {
```
Comment (mastercactapus) [inline web/src/app/users/UserPhoneNumberFilterContainer.js:43]: Since this is net-new let's make it tsx and use typescript types instead of propTypes

**Review by mastercactapus — CHANGES_REQUESTED** [review]

File: web/src/app/users/UserPhoneNumberFilterContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { useDispatch, useSelector } from 'react-redux'
+
+interface UserPhoneNumberFilterContainerProps {
+  anchorRef?: Ref<HTMLElement>
+}
+
+export default function UserPhoneNumberFilterContainer(
+  props: UserPhoneNumberFilterContainerProps,
+): JSX.Element {
+  const searchParam = String(useSelector(searchSelector))
```
Comment (mastercactapus) [inline web/src/app/users/UserPhoneNumberFilterContainer.tsx:17]: `searchSelector` always returns a string (default is `''`) so there's no need to cast, however, the types in `url.ts` are a bit ambiguous currently (should probably fix that) so we can just indicate the type for now
```
  const searchParam = useSelector(searchSelector) as string
```

However, a better option would be to use the hook for params directly:
```suggestion
  const [searchParam, setSearchParam] = useURLParam('search', '')
```

**Review by mastercactapus — CHANGES_REQUESTED** [review]
Can we get a quick cypress test for the happy path on this?

**Review by arurao — DISMISSED** [review]
LGTM

**Comment by mastercactapus** [comment]
I think this is okay without a verification server or fuzzy searching for initial implementation -- that can be subsequent work if there's interest.

If we can get the happy path tested with existing testing tools, conflicts resolved, and comments addressed it should be good to merge.

**Review by mastercactapus — APPROVED** [review]

**Review by mastercactapus — APPROVED** [review]

**Review by arurao — APPROVED** [review]

**Pull request merged**